homeassistant/components/unifi/sensor.py

```diff
@@ -425,7 +440,10 @@ def async_update_state(self, event: ItemEvent, obj_id: str) -> None:
         """
         description = self.entity_description
         obj = description.object_fn(self.controller.api, self._obj_id)
-        if (value := description.value_fn(self.controller, obj)) != self.native_value:
+        # Update the value only if value is considered to have changed relative to its previous state
+        if description.value_changed_fn(
+            self.native_value, (value := description.value_fn(self.controller, obj))
+        ):
             self._attr_native_value = value
```

Comment:
I've been thinking about this problem, wouldn't it be better to store the previous reported uptime value (without conversion) from the controller as that will remove any conversion issue between host time and controller time

Reply:
If I understand correctly, you mean storing the uptime value (in seconds) as the actual entity value? The problem there is that this is just a constantly-changing integer that would spam the recorder even more than the current bug does.

The core of the problem isn't that the host and controller have different clocks; It's that we received a number of "seconds ago" without knowing how stale that value is (and the staleness is unpredictable).

Reply:
Sorry, maybe I was unclear, I meant store it for reference, when state was last updated what was the raw value

Reply:
Sorry, I'm not connecting the dots; what does that buy us or how would the logic change? If I'm not mistaken, the fundamentals of the problem don't really change (since the `uptime` returned from the controller API has all of the uncertainty baked into it). The conversion between the `uptime` integer and our native `datetime` isn't contributing much there IMO

Reply:
IIRC there was some discrepancy in the logic that could trigger a difference in comparison when doing the utcnow - uptime check every 30 seconds due to slight difference in time between systems. If relevant diff comparison is done on the uptime value you remove one layer of uncertainty.

Reply:
I don't think so? Since you'd have to contrast both the previous uptime value _and_ your best approximation of the absolute time when you think the controller computed it, which leaves you with essentially the same amount of information as we're getting here (i.e. that we assume that `async_update_state` is being triggered as soon as possible after the coordinator's API call finishes).

> when doing the utcnow - uptime check every 30 seconds

Not that it changes the rationale, but in my observation we're getting updates from the coordinator once or twice every second (for every adopted Unifi device and the controller itself).

Reply:
I guess maybe it changes as you now compare on seconds as well, whereas before it was limited to minutes.